### ISSUE TITLE
Support for subtitle provider authentication

### ISSUE THREAD

**Issue by fortiko** [mdhiggins/sickbeard_mp4_automator#1410]
Hello! I might just have missed it in the documentation, but I cannot find how to specify username and password for certain subtitle providers (such as OpenSubtitles for instance).

Below you will find 2 small changes that would enable sickbeard_mp4_automator to simply pass the username and password specified in the autoProcess.ini to subliminal, based on this comment: https://github.com/Diaoul/subliminal/issues/848#issuecomment-349725997

```
[Subtitle]
download-providers = opensubtitles,tvsubtitles
; max 64 characters per password (subliminal restriction)
; see https://github.com/Diaoul/subliminal/issues/1008#issuecomment-758682064
download-providers-auth = {'opensubtitles': {'username': 'opensub_user', 'password': 'opensub_pass'},'tvsubtitles': {'username': 'test', 'password': 'test'}}
```

and these are the code changes required:

```
diff --git a/resources/mediaprocessor.py b/resources/mediaprocessor.py
index 28ce863..416c481 100644
--- a/resources/mediaprocessor.py
+++ b/resources/mediaprocessor.py
@@ -1318,7 +1318,7 @@ class MediaProcessor:
                     video.release_group = og.release_group
                     video.resolution = og.resolution

-                subtitles = subliminal.download_best_subtitles([video], languages, hearing_impaired=self.settings.hearing_impaired, providers=self.settings.subproviders)
+                subtitles = subliminal.download_best_subtitles([video], languages, hearing_impaired=self.settings.hearing_impaired, providers=self.settings.subproviders, provider_configs=self.settings.subproviders_auth)
                 saves = subliminal.save_subtitles(video, subtitles[video])
                 paths = [subliminal.subtitle.get_subtitle_path(video.name, x.language) for x in saves]
                 for path in paths:
diff --git a/resources/readsettings.py b/resources/readsettings.py
index a7b3a14..33b5732 100644
--- a/resources/readsettings.py
+++ b/resources/readsettings.py
@@ -1,6 +1,7 @@
 import os
 import sys
 import locale
+from ast import literal_eval

 try:
     from configparser import ConfigParser
@@ -753,6 +754,7 @@ class ReadSettings:
         self.downloadsubs = config.getboolean(section, "download-subs")
         self.hearing_impaired = config.getboolean(section, 'download-hearing-impaired-subs')
         self.subproviders = config.getlist(section, 'download-providers')
+        self.subproviders_auth = literal_eval(config.get(section, 'download-providers-auth'))
         self.embedsubs = config.getboolean(section, 'embed-subs')
         self.embedimgsubs = config.getboolean(section, 'embed-image-subs')
         self.embedonlyinternalsubs = config.getboolean(section, 'embed-only-internal-subs')

**Comment by mdhiggins** [comment]
In general I try not to include raw JSON in an INI config file, but I went ahead and did some updates to readsettings.py to handle parsing out these credentials

First, I decided to move the subliminal settings to its own subsection `[Subtitle.Subliminal]`
Second, I added another dynamic subsection `[Subtitle.Subliminal.Auth]` in the format `provider = user:pass`

```
[Subtitle.Subliminal]
download-subs = True
download-hearing-impaired-subs = False
providers = 

[Subtitle.Subliminal.Auth]
opensubtitles = osuser:ospass
tvsubtitles = tvuser:tvpass
```

This will be parsed and converted into a python dict that should look like this which will be passed to subliminal using the `provider_configs` parameter. This subsection is dynamic so any provider name can be specified in the event that new providers come along (not hard coded to known/existing providers).

```
{'opensubtitles': {'osuser': 'ospass'}, 'tvsubtitles': {'tvuser': 'tvpass'}}
```

The code includes auto-migration of the old settings to the new subluminal subsection for those 3 options so next time you start manual.py everything will be auto updated then you just need to add your credentials.

I've included these changes on the fancyfilters branch which includes a few other changes I plan on migrating to master soon
https://github.com/mdhiggins/sickbeard_mp4_automator/tree/fancyfilters

See if this branch works for you and let me know and I'll merge it in

d1a8f356392faba2bda3021868d6f71c40758b07
